src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIAttackRangedBowBarbarian.java

```diff
@@ -0,0 +1,196 @@
+    private final        int    ONE_MINUTE           = 60;
+    private final        int    TWENTY_SECONDS       = 20;
+    private final        double ONE_QUARTER_BLOCK    = 0.25D;
+    private final        float  HALF_BLOCK           = 0.5F;
+    private final        double THREE_QUARTERS_BLOCK = 0.75D;
+    private final        float  MAX_DEGREES          = 30.0F;
+    private static final int    MUTEX_BITS           = 3;
+
+    public EntityAIAttackRangedBowBarbarian(final AbstractArcherBarbarian archer, final double speedAmplifier, final int delay, final double maxDistance)
+    {
+        super();
+        this.entity = archer;
+        this.moveSpeedAmp = speedAmplifier;
+        this.attackCooldown = delay;
+        this.maxAttackDistance = maxDistance * maxDistance;
+        this.setMutexBits(MUTEX_BITS);
+    }
+
+    public void setAttackCooldown(final int coolDown)
+    {
+        this.attackCooldown = coolDown;
+    }
+
+    /**
+     * Returns whether the EntityAIBase should begin execution.
+     */
+    public boolean shouldExecute()
+    {
+        if (this.entity.getAttackTarget() == null)
+        {
+            return false;
+        }
+        return this.isBowInMainhand();
+    }
+
+    protected boolean isBowInMainhand()
+    {
+        return !this.entity.getHeldItemMainhand().isEmpty() && this.entity.getHeldItemMainhand().getItem() == Items.BOW;
+    }
+
+    /**
+     * Returns whether an in-progress EntityAIBase should continue executing
+     */
+    public boolean continueExecuting()
+    {
+        return (this.shouldExecute() || !this.entity.getNavigator().noPath()) && this.isBowInMainhand();
+    }
+
+    /**
+     * Execute a one shot task or start executing a continuous task
+     */
+    public void startExecuting()
+    {
+        super.startExecuting();
+        this.entity.setSwingingArms(true);
+    }
+
+    /**
+     * Resets the task
+     */
+    public void resetTask()
+    {
+        super.resetTask();
+        this.entity.setSwingingArms(false);
+        this.seeTime = 0;
+        this.attackTime = -1;
+        this.entity.resetActiveHand();
+    }
+
+    /**
+     * Updates the task
+     */
+    public void updateTask()
+    {
+        final EntityLivingBase entitylivingbase = this.entity.getAttackTarget();
+
+        if (entitylivingbase != null)
+        {
+            final double d0 = this.entity.getDistanceSq(entitylivingbase.posX, entitylivingbase.getEntityBoundingBox().minY, entitylivingbase.posZ);
+            final boolean flag = this.entity.getEntitySenses().canSee(entitylivingbase);
+            final boolean flag1 = this.seeTime > 0;
+
+            if (flag != flag1)
+            {
+                this.seeTime = 0;
+            }
+
+            if (flag)
+            {
+                ++this.seeTime;
+            }
+            else
+            {
+                --this.seeTime;
+            }
+
+            if (d0 <= (double) this.maxAttackDistance && this.seeTime >= TWENTY_SECONDS)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "double". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1905)